src/encoder.rs

```diff
@@ -160,11 +160,12 @@ pub struct Sequence {
 }
 
 impl Sequence {
-  pub fn new(config: &EncoderConfig) -> Sequence {
+  pub fn new(config: &EncoderConfig) -> Option<Sequence> {
```

Comment:
I'd keep the assert and move the check (using ilog maybe) where the configuration is validated. The sequence is an internal detail and should never fail no matter the input.